### PR TITLE
add io.github.rinigus.PureMaps

### PR DESCRIPTION
**Pull request by rinigus**
This is to add [Pure Maps](https://github.com/rinigus/pure-maps) to Flathub.

Pure Maps is a map application used on Sailfish OS for viewing & searching maps, finding routes, and navigation along them. When compared to Sailfish version, this version is stripped of online provider's API keys. This is to avoid running out of free quotas and effectively deny SFOS users access to maps. However, each user can specify personal API keys via Preferences.

While on normal desktop usage is limited to checking out maps, on Linux setups with GPS this can be used as a full blown navigation program.

The current version has 2 problems that I hope someone can help to resolve.

1. I don't know how to specify location permissions for flatpak. Since its a mapping application, this permission is needed. I couldn't test whether GPS location can be obtained on PC using Kde/5.11 platform either - don't have GPS hooked to PC.

2. For some reason, local builds fail if I enable `install -D packaging/pure-maps.appdata.xml /app/share/appdata/io.github.rinigus.PureMaps.appdata.xml` at the end of JSON with 
```
Renaming icon hicolor/256x256/apps/pure-maps.png to hicolor/256x256/apps/io.github.rinigus.PureMaps.png
Renaming icon hicolor/86x86/apps/pure-maps.png to hicolor/86x86/apps/io.github.rinigus.PureMaps.png
Renaming icon hicolor/128x128/apps/pure-maps.png to hicolor/128x128/apps/io.github.rinigus.PureMaps.png
Renaming icon hicolor/108x108/apps/pure-maps.png to hicolor/108x108/apps/io.github.rinigus.PureMaps.png
Rewriting contents of io.github.rinigus.PureMaps.desktop
Running appstream-compose
Processing application io.github.rinigus.PureMaps
Error loading desktop file: Failed to find icon: Failed to find icon io.github.rinigus.PureMaps
Error: ERROR: appstream-compose failed: Child process exited with code 1
```
The icons seem to be all there and I couldn't spot something strange in JSON or appdata XML. 

Instead of trying to resolve these issues, I am asking for review and help with this submission. Note that I'll be out of office for 2 days and my replies maybe delayed till that time.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build io.github.rinigus.PureMaps

**Comment by nedrichards** [comment]
I had a quick look at the [QtPositioning](https://doc.qt.io/qt-5.11/qtpositioning-index.html) docs and it seems that it uses GeoClue on Linux to get a position. So I'd assume that you'd need:
> "--system-talk-name=org.freedesktop.GeoClue2"
to get access to GeoClue.

**Comment by rinigus** [comment]
Its using GeoClue version 0.12.99 - not sure what that means. I can always try to add `--system-talk-name=org.freedesktop.GeoClue2`, but let's see if the bot will be happier than my local setup.

Taking into timing constrains, I'll probably get to it 2 days later ... getting late over here.

**Comment by rinigus** [comment]
Looks like bot failed on the same error. Any idea why it cannot find the icons?

**Comment by nedrichards** [comment]
@rinigus the other thing to do, is if you run your flatpak app with `--log-session-bus` and then `--log-system-bus` then you can see what address' you're trying to contact (you'll notice the large DENIED entries in the logs as the sandboxing refuses the connection).

**Comment by nedrichards** [comment]
As expected the build fails at 'cannot find icon' https://flathub.org/builds/#/builders/2/builds/6086 (which is good, means no other problems!)

**Comment by rinigus** [comment]
Thanks for logging tip, will try.

File: io.github.rinigus.PureMaps.json [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        "--talk-name=org.osm.scout.server1",
+        /* monitor availibility of OSM Scout Server */
+        "--talk-name=org.freedesktop.DBus",
+        "--socket=pulseaudio"
+    ],
+    "cleanup": ["/usr/include", "/usr/tests", "/usr/lib/libqmapboxgl.a", "/usr/lib/pkgconfig", "/opt"],
+    "rename-icon": "pure-maps", 
+    "modules": [
+        {
+            "name": "nemodbus",
```
Comment (barthalion) [inline io.github.rinigus.PureMaps.json]: Why not use qmake buildsystem here?

File: io.github.rinigus.PureMaps.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://git.merproject.org/mer-core/nemo-qml-plugin-dbus/-/archive/2.1.20/nemo-qml-plugin-dbus-2.1.20.tar.gz",
+                    "sha256": "d52b97d700ee17e3f20b1dcb84c58502d54d12fcec487a5da22061298f811075"
+                }
+            ]
+        },
+        {
+            "name": "pyotherside",
```
Comment (barthalion) [inline io.github.rinigus.PureMaps.json]: Why not use qmake buildsystem here?

File: io.github.rinigus.PureMaps.json [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                {
+                    "type": "git",
+                    "url": "https://github.com/rinigus/pkg-mapbox-gl-native.git",
+                    "branch": "sfos",
+                    "commit": "426263007b95480d30939648bd92bb6341f8395f"
+                }
+            ]
+        },
+        {
+            "name": "mapbox-gl-qml",
```
Comment (barthalion) [inline io.github.rinigus.PureMaps.json]: Why not use qmake buildsystem here?

**Comment by bilelmoussaoui** [comment]
You might need to use rename-icon in order to fix the icon issue.

**Comment by nedrichards** [comment]
bot, build io.github.rinigus.PureMaps

**Comment by rinigus** [comment]
I used rename icon already, so that's why its not clear what to do after that

**Comment by nedrichards** [comment]
I've seen a problem like this before...this looks suspiciously like https://github.com/flathub/net.pioneerspacesim.Pioneer/issues/1

**Comment by rinigus** [comment]
Looks like it is the same as the issue with https://github.com/flathub/net.pioneerspacesim.Pioneer/issues/1 . 

I have the same error when I build locally using build-dir:

```
appstream-compose --prefix . io.github.rinigus.PureMaps                                
WARNING: Metadata origin not set, using 'example'
Processing application io.github.rinigus.PureMaps
Error loading desktop file: Failed to find icon: Failed to find icon io.github.rinigus.PureMaps
```

(had to add .desktop to `id` before that to make it work). The versions of my local packages:

```
[I] dev-libs/appstream
     Available versions:  0.10.6-r1(0/4) ~0.11.5(0/4) ~0.12.2(0/4) {apt doc +introspection qt5 test}
     Installed versions:  0.10.6-r1(0/4)(18:31:47 10/10/18)(qt5 -apt -doc -test)
     Homepage:            https://www.freedesktop.org/wiki/Distributions/AppStream/
     Description:         Cross-distro effort for providing metadata for software in the Linux ecosystem

[I] dev-libs/appstream-glib
     Available versions:  0.6.13(0/8) ~0.7.9(0/8) {doc +introspection nls stemmer}
     Installed versions:  0.6.13(0/8)(18:59:25 22/12/17)(introspection nls -stemmer)
     Homepage:            https://people.freedesktop.org/~hughsient/appstream-glib/
     Description:         Provides GObjects and helper methods to read and write AppStream metadata

```

**Comment by nedrichards** [comment]
When building a flatpak it's the versions of the tools inside the flatpak sdk you use, rather than your local packages, in fact my base os doesn't have a compiler or dev tools at all.

**Comment by rinigus** [comment]
Thanks for pointing to the issue with Pioneer package. Looks like we were hit with https://github.com/hughsie/appstream-glib/issues/271 . By adding symlink from /app/share to /app/usr/share, I can build Pure Maps.

Please consider for inclusion. As for additional GPS permissions, we will find them as soon as someone will start using it with GPS device and it should be simple to fix, assuming that it requires extra permissions from DBus.

**Comment by nedrichards** [comment]
io.github.rinigus.PureMaps

**Comment by rinigus** [comment]
is bot missing ?

**Comment by rinigus** [comment]
@nedrichards - did you forgot to specify "bot" in the build command? at least it looks like the bot hasn't started building it yet

**Comment by nedrichards** [comment]
bot, build io.github.rinigus.PureMaps

yes - I can't type it seems ;-)

**Comment by rinigus** [comment]
Looks like bot is happy now. What's next?

**Comment by TingPing** [comment]
@nedrichards We have a geoclue portal now, has anybody tested it: https://github.com/flatpak/xdg-desktop-portal/releases/tag/1.1.0

It is reasonable to leave the permission until 1.2 is out, but maybe add a comment saying it will go away in the future.

File: io.github.rinigus.PureMaps.json [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                    "type": "archive",
+                    "url": "https://github.com/MycroftAI/mimic/archive/1.2.0.2.tar.gz",
+                    "sha256": "6adcc9911b09d6e9513add41ad9dfc0893ece277f556419869520a0f0708c102"
+                }
+            ]
+        },
+        {
+            "name": "pure-maps",
+            "buildsystem": "simple",
+            "build-commands": [
```
Comment (TingPing) [inline io.github.rinigus.PureMaps.json]: Your build system should handle all of this in a single `make install`. It doesn't make sense to have package specific workarounds.

**Comment by rinigus** [comment]
Re geoclue: looks like Qt 5.12 will have geoclue2 support (https://codereview.qt-project.org/#/c/229911/), which I expect is needed for the geoclue portal. So, we may require KDE/5.12 platform for it. 

Where I should add the corresponding comment?

**Comment by TingPing** [comment]
Hmm, I see you removed the permission so I guess a comment isn't needed. But I imagine somebody will open an issue if a feature isn't working.

**Comment by rinigus** [comment]
@TingPing: which permission are you talking about? Maybe it was an accident...

**Comment by TingPing** [comment]
https://github.com/flathub/flathub/pull/717#issuecomment-438027115

**Comment by rinigus** [comment]
OK, but Qt uses geoclue 0.12.99 => not geoclue2. At least, not until 5.12. So, this permission should not be of any help right now.

File: io.github.rinigus.PureMaps.json (hunk 4 above)
Comment (rinigus) [reply io.github.rinigus.PureMaps.json]: I moved packaging commands into Pure Maps Makefile, in line with the used targets for Sailfish OS RPM generation. Similar to SFOS, I added a new target and handle the specifics over there. Corresponding commits are published

**Comment by nedrichards** [comment]
Ah, OK - KDE runtime 5.12 is likely only a couple of weeks away though - may be worth holding this until that's available given how important location is to this app.

**Comment by nedrichards** [comment]
bot, build io.github.rinigus.PureMaps

**Comment by rinigus** [comment]
RE holding till 5.12: its a choice. Currently Pure Maps can be used as a map viewer / explorer, without GPS. To use it with GPS, you probably want to have some kind of portable device with it running Linux and, for this distribution model, with flatpak. I have no idea how many users are there with such portable setup, so it is hard to judge how many will be hit with this issue

I would suggest to proceed with review and inclusion. As soon as 5.12 is available, I can test and update it to that platform.

**Comment by barthalion** [comment]
Yeah, I agree there's no point in waiting. I'll merge it tomorrow morning if nobody beats me to it.

**Comment by rinigus** [comment]
Thank you!

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/io.github.rinigus.PureMaps

**Pull request closed without merging**